js/src/data.ts

```diff
@@ -177,16 +180,16 @@ export class Data<T extends DataType = DataType> {
                 // if we have a nullBitmap, truncate + slice and set it over the pre-filled 1s
                 if (this.nullCount > 0) {
                     nullBitmap.set(truncateBitmap(offset, length, this.nullBitmap), 0);
+                    Object.assign(this, { nullBitmap });
```

Comment:
If we assign null count here, the update to the null count will be wrong. Also, we know that the null count has not yet changed so there is no need to set it here.